### PR TITLE
harvester: decouple plot refresh from API requests

### LINKED ISSUE #5208: [BUG] harvester.refresh_plots() should not be blocking?
I see long delays, getting "Looking up qualities" messages in the debug.log.  I read through the other issues about similar long lookups, and it seemed like the delays increase with the size of the farm.  Which is to say -- the reported delays are not actually about a specific hard drive lookup.  

In this file: `/chia/harvester/harvester_api.py`
every time there's a challenge, the harvester checks to see if it should refresh all the plots.

```
# LINE ~ 72
# Refresh plots to see if there are any new ones
if start - self.harvester.last_load_time > self.harvester.plot_load_frequency:
	await self.harvester.refresh_plots()
	self.harvester.last_load_time = time.time()
```

Is it necessary to use `await` for the `self.harvester.refresh_plots()` call?  Isn't that going to block everything subsequent in the loop?

I have over 1800 plots, and many of them are on NAS connections (I know, that's apparently not ideal) but the subequent error reports keep telling me that a whole batch of plot lookups took too long.  

```
# LINE ~ 196
# Concurrently executes all lookups on disk, to take advantage of multiple disk parallelism
total_proofs_found = 0
for filename_sublist_awaitable in asyncio.as_completed(awaitables):
	filename, sublist = await filename_sublist_awaitable
	time_taken = time.time() - start
	if time_taken > 5:
		self.harvester.log.warning(
			f"Looking up qualities on {filename} took: {time_taken}. This should be below 5 seconds "
			f"to minimize risk of losing rewards."
		)
```

So, it's good that the lookups are happening in parallel, but this takes place AFTER the blocking `self.harvester.refresh_plots()` call at the top of the process.  

I don't know what the solution is.  Can `refesh_plots()` run in its own thread, divorced from the lookup loop?  At minimum, the log files should say something like:

"Probably this lookup itself was just fine, but the system-wide collection of refesh_plots() pushed your response time up to XXX seconds."

### PR DESCRIPTION
**Pull request by ctrlaltdel**
Moved plot refresh in its own asynchronous task, this helps prevent slow
responses on big harvesters (5K+ plots) when the plot refresh is triggered.

Fixes #5208.

### CODE REVIEW

File: chia/harvester/harvester_api.py [hunk 1]
```diff
@@ -82,9 +77,7 @@ def blocking_lookup(filename: Path, plot_info: PlotInfo) -> List[Tuple[bytes32,
             try:
                 plot_id = plot_info.prover.get_id()
                 sp_challenge_hash = ProofOfSpace.calculate_pos_challenge(
-                    plot_id,
```
Comment (mariano54) [inline chia/harvester/harvester_api.py:85]: Can you revert these automatic style changes to keep the diff small?

File: chia/harvester/harvester.py [hunk 2]
```diff
@@ -117,6 +120,15 @@ async def refresh_plots(self):
         if changed:
             self._state_changed("plots")
 
+    async def _periodically_refresh_plots_task(self):
+        while True:
+
+            # Don't run plot refresh until farmer handshake was done
+            if self.farmer_public_keys and self.pool_public_keys:
```
Comment (mariano54) [inline chia/harvester/harvester.py:127]: If we start the refresh before handshake is done, we might wait a long time to load the plots. We should load the plots immediately after the handshake. Can we maybe have another loop before this while True?

**Comment by mariano54** [comment]
Thanks for the change! made some comments

**Review by wjblanke — DISMISSED** [review]
aok

File: chia/harvester/harvester.py (hunk 2 above)
Comment (wjblanke) [reply chia/harvester/harvester.py]: Hmm still some odd behavior going on with the CI tests

**Comment by HarukaMa** [comment]
Looks like the farming is still being blocked - the harvester would just wait for scanning to finish before starting to process SPs. Those SPs received during the scan will still be delayed.

Log looks like this:

```
2021-06-18T03:58:47.318 harvester chia.harvester.harvester: INFO     ** plots were eligible for farming 4cac0ca524... Found 0 proofs. Time: 6.27090 s. Total ***** plots
2021-06-18T03:58:47.319 harvester chia.harvester.harvester: INFO     ** plots were eligible for farming 56b44ec77f... Found 0 proofs. Time: 4.15971 s. Total ***** plots
2021-06-18T03:58:47.730 harvester chia.harvester.harvester: INFO     ** plots were eligible for farming 56b44ec77f... Found 0 proofs. Time: 2.67811 s. Total ***** plots
```

**Pull request closed without merging**